app/Controllers/PostControllers.php

```diff
@@ -0,0 +1,318 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\Controllers;
+
+
+use App\View;
+use App\Models\Post;
+use App\Models\User;
+use App\Models\Comment;
+
+class PostControllers
+{
+    CONST REQUEST_METHOD = 'POST';
+    protected Post $postModel;
+    protected User $userModel;
+    protected Comment $commentModel;
+
+
+
+    public function __construct()
+    {
+        $this->postModel = new Post();
+        $this->userModel = new User();
+        $this->commentModel = new Comment();
+    }
+
+
+    /**
+     * function to get All Blog Post Data.
+     * @return View
+     */
+    public function blogPosts(): View
+    {
+        $posts = $this->postModel->getAllBlogPosts();
+        $data = [
+            'posts' => $posts,
+        ];
+        return View::make('/blogs/home', $data);
+    }
+
+
+    /**
+     * function for admin to see blog posts in admin homepage
+     * @return View
+     */
+    public function adminSeeBlogs(): View
+    {
+        $posts = $this->postModel->adminBlogInfoHome();
+        $blogData = [
+            'posts' => $posts
+        ];
+        return View::make('/admin/home', $blogData);
+    }
+
+
+    /**
+     * function to validate blog Data
+     * @param $data
+     * @return void
+     */
+    public function blogValidation($data): void
+    {
+        if (empty($data['title'])) {
+            $data['title_err'] = 'Please enter a blog title';
+        }
+        if (empty($data['blog_body'])) {
+            $data['blog_body_err'] = 'Please enter a blog body';
+        } elseif (strlen($data['blog_body']) > 76) {
+            $data['blog_body_err'] = 'Blog body must be less than 76 characters';
+        }
+    }
+
+    /**
+     * function to add Blog and make sure all input areas are filled in
+     * @return View|void
+     */
+    public function addBlog()
+    {
+        if ($_SERVER['REQUEST_METHOD'] == self::REQUEST_METHOD) {
+            $_POST = filter_input_array(INPUT_POST, FILTER_SANITIZE_STRING);
+            $data = [
+                'title' => trim($_POST['title']),
+                'blog_body' => trim($_POST['blog_body']),
+                'user_id' => $_SESSION['user_id'],
+                'title_err' => '',
+                'blog_body_err' => '',
+            ];
+            $this->blogValidation($data);
+            if (empty($data['title_err']) && empty($data['blog_body_err'])) {
```

Comment:
I think you can move this if statement in the blogValidation function because it's still being repeated in the other functions.